src/searcher/mod.rs

```diff
@@ -349,11 +351,62 @@ impl Searcher {
         Ok(())
     }
 
+    pub fn search_slice_callback_no_path(
+        &mut self,
+        query_context: QueryContext,
+        slice: RopeOrSlice,
+        // slice: impl TextProvider<'a> + Parseable + 'a,
+        tree: Option<&Tree>,
+        mut callback: impl FnMut(CaptureInfo),
+    ) -> Result<(), ConfigError> {
+        self.check_config()?;
+
+        log::trace!("slice reader: searching via multiline strategy");
+        let mut query_cursor = QueryCursor::new();
+        let query = &query_context.query;
+        let capture_index = query_context.capture_index;
+        let filter = &query_context.filter;
+        let tree: Cow<'_, Tree> = tree.map_or_else(
```

Comment:
This still feels a bit "ceremonial" but I think this is actually a sensible "idea" for how to unify the types of "we got passed an `Option<&T>` (where we don't want to `.clone()` the inner `&T`) and we either unwrap that or have the ability to create a new (owned) `T`"

(I know that in the past I have done clunkier things in such situations)